src/DiagnosticsDialog.cpp

```diff
@@ -164,3 +164,5 @@ void DiagnosticsDialog::copyToClipboard()
     QClipboard* clipboard = QApplication::clipboard();
     clipboard->setText(fullLog);
 }
+
+DiagnosticsDialog* globalDiagnosticsDialog = NULL;
```

Comment:
make this a proper documented singleton, static member variables inside DiagnosticsDialog, remove the variable from MainWindow, make everything use it via the singleton var.

Reply:
I have no objection to document it, but I think that the rest is out of scope for this PR. The need for the singleton comes from tight coupling between the scanning logic and UI. As this coupling causes another problems, I would rather spend some time by decoupling that than making this fix nicer.
I see this as a matter of discussion, but even if we decide to use the singleton and make the whole code use it, it would fit to another PR perfectly.

Reply:
I have tried the signal approach before, but I have failed - I am a QT signal newbie and there was nobody to ask at that time: Objects can send signals to the dialog thanks to the fact that they are connected to it (see src/MainWindow.cpp:630). MainWindow can connect anything, but there is no way how to present instances of remediation savers to it, ResultsViewer could connect something, but it doesn't know of the diagnostics dialog.
I have not explored the signal way further, because there seemed to be a better way.

Reply:
I can help you with this, the idea is to signal errorMessage with a string and connect that. Passing the saver ref shouldn't be necessary.

Reply:
Check out https://github.com/OpenSCAP/scap-workbench/blob/v1-1/src/MainWindow.cpp#L634

Reply:
We have agreed that this way of solving the problem by passing signals is indeed not an improvement, so properly documenting the global variable will be enough for the time being.